include/integration/LV_MC_integration.hpp

```diff
@@ -0,0 +1,133 @@
+#include "ode_solvers/oracle_functors.hpp"
+#include "random_walks/random_walks.hpp"
+#include "volume/volume_sequence_of_balls.hpp"
+#include "volume/volume_cooling_gaussians.hpp"
+#include "volume/volume_cooling_balls.hpp"
+#include "misc.h"
+
+typedef double NT;
+typedef Cartesian<NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef std::vector<Point> Points;
+typedef HPolytope<Point> HPOLYTOPE;
+typedef boost::mt19937 RNGType;
+typedef BoostRandomNumberGenerator<RNGType, NT> RandomNumberGenerator;
+
+typedef const unsigned int Uint;  // Positive constant value for no of samples & dimensions
+enum volumetype { CB ,CG ,SOB }; // Volume type for polytope
+
+
+template
+<
+    typename EvaluationFunctor,
+    typename GradientFunctor,
+    typename WalkType,
+    typename Polytope = HPOLYTOPE,
+    typename Point,
+    typename NT
+>
+NT lovasz_vempala_integrate(EvaluationFunctor &g,
+                            GradientFunctor &grad_g,
+                            Polytope &P,
+                            Point x0,
+                            NT B,
+                            unsigned int walk_length = 10,
+                            NT epsilon = 0.1)
+{
+    unsigned int n = P.dimension();
+    unsigned int m = (unsigned int) ceil(sqrt(n) * log(B));
+    unsigned int k = (unsigned int) ceil(512 / pow(epsilon,2) * sqrt(n) * log(B));
+    std::cout << "n = " << n << " m = " << m << " k = " << k << std::endl;
+
+    NT volume = volume_sequence_of_balls <BallWalk, RandomNumberGenerator, Polytope> (P, epsilon, walk_length);
+    NT alpha = (NT) 1 / B;
+    NT alpha_prev = 0;
+    NT log_W = log(volume);
+    NT W_current = (NT) 0;
+
+    RandomNumberGenerator rng(1);
+
+    // Initialize HMC walks using OptimizationFunctor <CustomFunctor::GradientFunctor , CustomFunctor::FunctionFunctor>
+    typedef OptimizationFunctor::GradientFunctor
+      <Point, EvaluationFunctor, GradientFunctor> NegativeGradientOptimizationFunctor;
+    typedef OptimizationFunctor::FunctionFunctor
+      <Point, EvaluationFunctor, GradientFunctor> NegativeLogprobOptimizationFunctor;
+    typedef LeapfrogODESolver<Point, NT, Polytope, NegativeGradientOptimizationFunctor> Solver;
+
+    OptimizationFunctor::parameters<NT, EvaluationFunctor, GradientFunctor> opt_params(1, n, g, grad_g);
+
+    NegativeLogprobOptimizationFunctor f(opt_params);
+    NegativeGradientOptimizationFunctor F(opt_params);
+      
+    HamiltonianMonteCarloWalk::parameters <NT, NegativeGradientOptimizationFunctor> hmc_params(F, n);
+
+    HamiltonianMonteCarloWalk::Walk
+      <Point, Polytope, RandomNumberGenerator, NegativeGradientOptimizationFunctor, NegativeLogprobOptimizationFunctor, Solver>
+      hmc(&P, x0, F, f, hmc_params);
+
+    // Check and evaluate for all samples breaks when variance > 1, i.e. a > 1
+    for (int i = 1; i <= m && alpha < 1; i++ ) {
+
+        alpha *= (1 + 1 / sqrt(n)); // variance sequence algorithm stops when variance > 1
```

Comment:
Here you should use `f.set_variance(alpha)` and `F.set_variance(alpha)`. Also I would prefer using `f` and `grad_f` as variable names to that we do not confuse them for something else (since we also have `g` and `grad_g` variables)